model_signing/signature/signing.py

```diff
@@ -0,0 +1,209 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This package provides the functionality to sign models"""
+
+from typing import Optional
+
+import abc
+
+from absl import logging as log
+from cryptography.hazmat.primitives import serialization as crypto_serialization
+from cryptography.hazmat.primitives.asymmetric import ec as crypto_ec
+from cryptography.hazmat.primitives.hashes import SHA256
+from cryptography import x509 as crypto_x509
+from google.protobuf import json_format
+from in_toto_attestation.v1 import statement
+from sigstore import dsse
+from sigstore import oidc
+from sigstore import sign
+from sigstore_protobuf_specs.dev.sigstore.bundle import v1 as bundle_pb
+from sigstore_protobuf_specs.dev.sigstore.common import v1 as common_pb
+from sigstore_protobuf_specs.io import intoto as intoto_pb
+
+from signature import utils
+
+
+class Signer(abc.ABC):
+    """Signer is the abstract base class for all signing methods."""
+
+    @abc.abstractmethod
+    def sign(self, stmnt: statement.Statement) -> bundle_pb.Bundle:
```

Comment:
Let's not tie in too much with sigstore or in-toto types in the abstract class. We should be able to support any type of serialization of the model, not only in-toto

Reply:
Still very much on the fence with this one since I see a lot of usability/ease of use advantages with a stricter regime. 
Advantage right now we have this datatype available while the manifest isn't here yet.

Reply:
I think manifest support is on its way https://github.com/sigstore/model-transparency/pull/190?